.eslintrc.js

```diff
@@ -39,85 +19,223 @@ module.exports = {
+            pathGroupsExcludedImportTypes: ["react"],
+            alphabetize: {
+              order: "asc",
+              caseInsensitive: true,
+            },
+          },
+        ],
+
+        // Rules for React Hooks.
+        "react-hooks/rules-of-hooks": ERROR, // Check rules of Hooks
+        "react-hooks/exhaustive-deps": WARN, // Warn useMemo, useEffect dependencies
+
+        // Rules for React.
+        "react/destructuring-assignment": OFF, // Allow use of dot notation, for example user.id (airbnb rule)
+        "react/display-name": OFF,
+        "react/function-component-definition": [
+          errorIfStrict,
+          {
+            namedComponents: "arrow-function", // Allow named components with arrow functions (airbnb rule)
+          },
+        ],
+        "react/jsx-filename-extension": [1, { extensions: [".tsx"] }], // Allow JSX in TSX file (airbnb rule)
+        "react/jsx-props-no-spreading": [
+          errorIfStrict,
+          {
+            custom: "ignore",
+            explicitSpread: "ignore",
+          },
+        ],
+        "react/prop-types": OFF, // (airbnb rule)
+        "react/react-in-jsx-scope": OFF, // Disable because there is no need to import React in React 17+ (airbnb rule)
+        "react/require-default-props": OFF, // Allow not setting defaults for props (airbnb rule)
+        "react/sort-comp": [
+          errorIfStrict,
+          { order: ["everything-else", "render"] },
+        ],
+        "react/no-unstable-nested-components": OFF, // This rule should be removed as part of EVG-17265.
+      },
+      parserOptions: {
+        project: ["./tsconfig.json"],
+      },
+    },
+    // For Jest files.
+    {
+      files: ["src/**/*.test.ts", "src/**/*.test.tsx"],
+      extends: ["plugin:testing-library/react", "plugin:jest/all"],
+      rules: {
+        "@typescript-eslint/unbound-method": OFF,
+        "jest/max-expects": [
+          WARN,
+          {
+            max: 10,
+          },
+        ],
+        "jest/no-hooks": OFF,
+        "jest/no-mocks-import": OFF,
+        "jest/prefer-expect-assertions": OFF,
+        "jest/unbound-method": OFF,
+      },
+    },
+    // For Storybook files.
+    {
+      files: ["src/**/*.stories.@(ts|tsx|js|jsx|mjs|cjs)"],
+      extends: ["plugin:storybook/recommended"],
+      rules: {
+        "storybook/story-exports": ERROR,
+        "storybook/no-stories-of": ERROR,
+        "storybook/no-redundant-story-name": WARN,
+        "storybook/csf-component": WARN, // Enables support for auto generated prop controls and documentation
+      },
+    },
+    // For Cypress files.
+    {
+      files: ["cypress/**/*.ts"],
+      extends: ["plugin:cypress/recommended"],
+      parserOptions: {
+        project: "cypress/tsconfig.json",
+      },
+    },
+    // For GraphQL files.
     {
-      files: ["*.graphql"],
-      parser: "@graphql-eslint/eslint-plugin",
-      plugins: ["@graphql-eslint"],
+      files: ["src/**/*.graphql"],
```

Comment:
It looks like the typescript rules are being applied to the graphql files.
![image](https://user-images.githubusercontent.com/4605522/229622196-c8676e8c-55c8-4038-95d3-541e1338a4ce.png)

Reply:
I couldn't quite replicate this but I added the sdlschema folder to the `.eslintignore` file, let me know if that worked!